src/DungeonMap.cpp

```diff
@@ -311,21 +319,26 @@ void CDungeonMap::FillArea( const Uint8 type, JRect *rcFill, const int direction
             GetTile( vCurPos )->SetFlags( DUNG_FLAG_LIT );
         }
     }
+}
 
-    // you still filled rcFill squares, just that one of them was a door.
-    printf( "filled from <%d %d> to <%d %d>\n", RECT_EXPAND( *rcFill ) );
-
-    if( bIsHallway )
-    {
-        // GetTile(*vOrigin)->SetType(DUNG_IDX_OPEN_DOOR);
-        GetTile( vOrigin )->SetType( DUNG_IDX_FLOOR );
-        // GetTile(*vOrigin)->SetFlags(~DUNG_FLAG_LIT);
-    }
-    else if( type != DUNG_IDX_WALL )
+void CDungeonMap::lightArea( const Uint8 type, JRect &rcLocal, JRect &rcEdges, JIVector &vOrigin )
```

Comment:
Light area does not need type, if you want to get functional, functions are atomic, they do one thing. This should not need origin either. Just the rect to light.